### ISSUE TITLE
Spike Growth balance

### ISSUE THREAD

**Issue by mlenser** [mlenser/kryx-rpg-issues#173]
Moving here for @itamarcu

> Currently this spell seems incredibly underpowered:
>
> [Spike Growth](https://www.kryxrpg.com/themes/spells/spike-growth) costs 2 mana to create a 3-meter radius sphere (slightly smaller than 4x4 squares on a grid). Creatures that move within it take... 1d4 per 1.5 meters, and it's difficult terrain, and potentially invisible the first time you cast it. So, if you get very lucky and somehow manage to put it in a hallway between yourself and a dozen goblins who now have no choice if they're chasing you and must pass through the spikes, they take... 4d4 damage each, as they pass through the spikes. that's really minor.
>
> Sure, the spell has the technical potential to cause stupid amounts of damage if you somehow grapple the creature and drag it back and forth through the spikes while also not stepping on them yourself and keeping in mind that it's difficult terrain (so even forcing the creature to use its movement to walk through the spikes is only 3d4 damage).... this is a *very* situational combo and the damage does not justify its high mana cost and tiny area of effect, if you ask me. If you're concerned about some odd combo that lets this generate 50 damage per turn, I'd rather you just buff the spell again and nerf that particular interaction, such as putting a limit on the amount of times a creature can get hit by the spikes in a single turn.
>
> Furthermore, this spell doesn't do any instant damage - and if you put it under an enemy, you're actually losing out on potential damage, as they need to cross less spikes to reach you. *Furthermore*, this just changes the floor - it doesn't affect flying creatures, hovering creatures, jumping creatures, etc.
>
> I'm pretty sure you'd rather have a Wall of Fire almost always, as it deals more damage (4d6, save for half), deals it instantly when you cast it, can't be jumped above, deals a damage type that is more resisted than standard piercing damage, and can block a longer section between two places (since if you can't even block the whole path to you, enemies will just walk around the spell). IF you really want to slow down your enemies with difficult terrain, use Erupting Earth for 5d6 instant damage, or use Fissures for the very big area and the potential to knock people prone (to slow them further), both of which don't require your valuable Concentration.

**Comment by DalenWBrauner** [comment]
The advantage of having it persist as a hazard for some time is comparable
to Fire Wall to be sure. I think that the hidden spikes is a cool
advantage, and I think you highlighted the perfect use case- you see a mob
of lower-level enemies approaching, so you slap down some spikes. This
should either reduce their HP, slow their approach, or both.

Wall of Fire does a similar thing, and so does another spell worth
comparing:
https://www.kryxrpg.com/themes/spells/guardian-of-faith

These spells are all very different, but they serve the same mechnical
purpose: area denial. And they have different ways of dealing with them:
Fire Wall and Guardian both can be seen, thereby and avoided (if possible)
whereas someone must somehow detect Spike Growth before they can avoid it.
But Spike Growth is easier to avoid once detected, by flying/jumping etc.
Similarly, Fire Wall and Guardian can both be saved against, so only half
the damage is guaranteed- Spike Growth always guarantees its damage, but
you (or someone nearby) only has to take a single instance of it (instead
of a full burst) before you recognize there's something to be avoided.

My argument is that these spells are very comparable functionally, and
already have very good tradeoffs for each other, in every respect except
for one: Damage.
For this spell to be useful in comparison to similar effects at this mana
cost, I think it needs to deal d6s.

**Comment by mlenser** [comment]
Spike growth is on [the spreadsheet](https://docs.google.com/spreadsheets/d/1VVduALSLGnbAlxJaFncqAxVMM_jun3eXhZeLScGOSvc/edit#gid=1121938024). It assumes 20 feet are moved, for ~4d4 each turn.


Here is spike growth before I've oticed that the difficult terrain is not taken into account:
![image](https://user-images.githubusercontent.com/235857/81320061-de732580-9090-11ea-91b4-d9cbc53d8a77.png)

With difficult terrain taken into account it's much too strong:
![image](https://user-images.githubusercontent.com/235857/81320323-4a558e00-9091-11ea-9c24-932b7902e927.png)

**Comment by mlenser** [comment]
> I think what the sheet doesn't consider is that the creature will rarely take that damage every turn, since it probably won't walk in circles inside the area.

The same is infinitely more true for every wall that does damage. Walls you walk through once. Spike Growth can last multiple rounds with the slow.

**Comment by DalenWBrauner** [comment]
The more I study the sheet and how this is calculated, the more I think this spell has a unique problem. It both:
- Only affects creatures on the ground
- Cannot forcibly affect a creature for the "full" or even "half" amount (as opposed to walls, which can be spawned on top of creature)

It's a spell that, unlike the walls we're comparing it to, can be avoided for almost the full amount of damage if:
- The creature has any jumping abilities, hovering abilities, wall/ceiling climb or flight speeds
- The creature does not need to move across the entire diameter

In other words, while a creature would normally save against a Wall's damage, instead a creature here has a lot more options to avoid the majority of damage from this spell and still cross the "sphere."

**Comment by DalenWBrauner** [comment]
I think this might calculate out to only a meager 5ft's worth (1d4) of damage being avoided on average.
Maybe the first 5ft dealt to a given creature should deal an extra 1d4 of damage?

**Comment by mlenser** [comment]
Spike Growth damage changed from 1d4 to 1d6 as the difficult terrain was miscalculated

**Comment by mlenser** [comment]
> Only affects creatures on the ground

90%+ of monsters

> Cannot forcibly affect a creature for the "full" or even "half" amount (as opposed to walls, which can be spawned on top of creature)

Walls only deal damage once when moved through and do not have difficult terrain

> The creature has any jumping abilities, hovering abilities, wall/ceiling climb or flight speeds

Requires seeing it placed

> The creature does not need to move across the entire diameter

The exact same is true for walls. These kind of spells are all about positioning.

**Comment by mlenser** [comment]
Closing as damage has been slightly buffed

**Comment by DalenWBrauner** [comment]
I see that it's closed and appreciate the changes, but I'll leave with:

> Requires seeing it placed
Not true. Requires stepping 5ft or observing someone else take damage

> 90%+ of monsters
which most can long jump if they know it's there

**Comment by mlenser** [comment]
> Not true. Requires stepping 5ft or observing someone else take damage

It doesn't become visible once thye take damage. Something pokes them in the foot. It's still camouflaged and the area is unknown.

> which most can long jump if they know it's there

That's pretty metagamey. How do they know the spell? How do they know how big the area is? How do they know where to jump?

**Comment by DalenWBrauner** [comment]
> That's pretty metagamey
Hrmm, very good point. I was thinking entirely from that perspective.

**Comment by Paulorpribeiro** [comment]
It's metagamey because this spell, imo, gives no clue of the hazard. The text reads 

> 
The transformation of the ground is camouflaged to look natural. Any creature that can’t see the area at the time the spell is cast must make a Perception check against your spell save DC to recognize the terrain as hazardous before entering it.
> 
As I understand, visually it's a regular patch of grass. Even if you see a bearded man speaking some weird words, the spell has no visual effects. And when you first take damage from the spell, sure you can stop moving to avoid further damage, but you don't know you are in a 2x2 square of hazardous terrain,  and you shouldn't  jump it unless you can somehow detect magic.

**Comment by itamarcu** [comment]
Thanks for buffing it a bit, and now I realized that this camouflage is indeed pretty good if your ruling is that a creature doesn't know where the spell starts or ends even after getting damaged by it.

It really depends on the GM, though, if monsters will just walk all the way across it (taking the full 4d6 damage) or stop after only 5 feet of movement, the moment they get damaged.

**Comment by mlenser** [comment]
> if your ruling is that a creature doesn't know where the spell starts or ends even after getting damaged by it.

That's what the RAW spell says - I haven't changed that aspect.

___________________

> It really depends on the GM, though, if monsters will just walk all the way across it (taking the full 4d6 damage) or stop after only 5 feet of movement, the moment they get damaged.

If they move across it and take the full damage then it's functionally better than Wall of Fire as there is no save and there is also a slow.
If they take the damage and stop then they've stopped on their approach to get where you don't want them to be so it has served its purpose.
